src/shared/dissect-image.c

```diff
@@ -1004,12 +1003,12 @@ int dissect_image(
                                 if (!sd_id128_is_null(root_uuid) && !sd_id128_equal(root_uuid, id))
                                         continue;
 
-                                designator = PARTITION_ROOT;
-                                architecture = native_architecture();
+                                assert((architecture = gpt_partition_type_uuid_to_arch(type_id)) >= 0);
```

Comment:
this is not safe. assert() may be optimized away via NDEBUG. If so the whole assignment is gone.

either use assert_se() (which is an assert where the expression can't allowed to be optimized away, because it has side-effects (hence the "se" in the name), like here. Or do it in two lines.